### ISSUE TITLE
Make feature definitions require a human readable description.

### ISSUE THREAD

**Issue by cyphactor** [codebreakdown/togls#5]
I think it is crucial that from a long term maintenance standpoint of feature toggles we have some human readable description explain what that particular feature toggle is/means.

**Comment by cyphactor** [comment]
So, I guess I am saying we should make the gem more opinionated and only let features be defined if a human readable description is provided. @BRIMIL01 or @RyanHedges have any thoughts?

**Comment by RyanHedges** [comment]
What's the use case? I call something like the following to get some information about the feature I'm working with from inside my code?

```ruby
Togls.feature.(:pop_up_login_form).description
#=> "Provides a pop up for logging into our system instead of sending to login page"
```

Or would it be something you'd use from a command line?

```terminal
$ Togls describe --all

Pop up login form:    ON
                      Provides a pop up for logging into our 
                      system instead of sending to login page.

Send follup email:    OFF
                      Sends an email after confirming xyz action.
```

I guess it would make sense because you might have 100's of features that you'd want to verify.

**Comment by cyphactor** [comment]
The use case I am implementing now related to this is as follows:

```ruby
Togls.feature(:pop_up_login_form, "control if pop up login form is used instead of normal login form").on
```
The following will be mechanism for accessing an individual features description.

```ruby
Togls.feature(:pop_up_login_form).description
```

The following will be a method that will return all the defined features, feature objects.

```ruby
Togls.features
```

The following is going to be a provide rake task that will display all the features, their description, and their status.

```shell
rake togls:features
```

**Comment by cyphactor** [comment]
From an outermost perspective this really results in the following interfaces:

1. defining a feature toggle with its **required** description

    ```ruby
    Togls.features do
      feature(:pop_up_login_form, "control if pop up login form is used instead of the normal one").on
    end
    ```
2. outputting all of the feature toggles including their status, key, and description via rake task

    ```shell
    rake togls:features
    ```

**Comment by cyphactor** [comment]
This was resolved by pull request #6 so I am closing this issue.
